linode/manager
# feat: [M3-7114] - Allow user to choose resize migration type when resizing Linode

## Description 📝
This work will introduce `warm` migrations which combine the benefits of `cold` and `live` migrations to greatly reduce customer downtime when resizing Linodes.

> **Important**
The notification and event copy/messaging for `warm` migrations will be updated with API work, but you test the UI flow, payload, feature flags and everything else. I will have to update the cypress tests later with actual copy (9/27)

## Major Changes 🔄

- [x] Add feature flag `unifiedMigrations`

**Resize Flow:**
- [x] Add an optional `migration_type` POST field for the `linode/instances/.../resize` endpoint that should be `warm` or `cold`. Backend defaults to `cold`, but in the UI we are defaulting to `warm` unless the user chooses otherwise.
- [x] Add a radio button group for selecting `warm` or `cold` migrations in the `LinodeResize` flow.
    - [x] If the Linode is powered off, we disable controls (even though warm is selected by default) and proceed with a `cold` migration.
- [x] Update & add cypress E2E tests for changes

**Linode Landing:**
- [x] As a user the flag icon is ambiguous and doesn't give much meaning, therefore we want to remove the icon altogether and provide and updated status where it makes sense. For resizes, we want to change the wording from `running` to `migrating` with a green dot icon


**Other:**
- [x]  Fixed the animations for `<Notice />` and fixed a bug in our `fadeIn` keyframes. _Pesky colons!_
    - [x]  In addition, the padding for the `<Notice />` didn't seem right or match `<DismissibleBanner />` which seemed to be the intention, so this has been fixed.

## TODO:
- [ ] Waiting on APIv4 dev work with event/notification messaging to update Cypress tests
- [x] Waiting on UX copy for messaging

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-09-13 at 9 30 01 PM](https://github.com/linode/manager/assets/125309814/f9c0bba4-6230-41b7-a649-8221c5342d39) | ![Screenshot 2023-09-13 at 9 29 52 PM](https://github.com/linode/manager/assets/125309814/ea9e5c2d-0da4-4522-8d2d-85fd05cd3e02) |

|Event Notification|
| ------- | 
|<img width="839" alt="Screenshot 2023-09-15 at 11 56 01 AM" src="https://github.com/linode/manager/assets/125309814/962d93fd-207c-4747-b0ee-05b8a489f751">|

| Enabled  | Disabled   |
| ------- | ------- |
| ![Screenshot 2023-09-28 at 9 29 14 AM](https://github.com/linode/manager/assets/125309814/6efa1bba-9684-447d-b052-8dc6945b4189) | ![Screenshot 2023-09-28 at 9 27 39 AM](https://github.com/linode/manager/assets/125309814/6b97fb88-ba18-47ae-a05c-12e10b9a248c) |

| Payload  |
| ------- |
|![Screenshot 2023-09-13 at 3 49 32 PM](https://github.com/linode/manager/assets/125309814/c6f4eb67-c158-4f66-b898-d12bc8f17fb3)|

| Linode Landing  |
| ------- |
|![Screenshot 2023-09-27 at 10 34 27 PM](https://github.com/linode/manager/assets/125309814/d7661f27-6062-4376-be4a-5806ca849283) ![Screenshot 2023-09-27 at 10 30 26 PM](https://github.com/linode/manager/assets/125309814/83326319-2a8a-4aa1-94ab-18b0917051db)|

| Feature Flag  |
| ------- |
|![Screenshot 2023-09-25 at 7 45 22 PM](https://github.com/linode/manager/assets/125309814/055af4f4-5540-49ee-a2be-d1053783b82c)|


## How to test 🧪
1. Enable `Unified Migrations` feature flag via dev tools
2. Go to Linodes landing page and resize a Linode from the action menu
3. Observe you now have a radio group to choose `warm` or `cold` migrations with `warm` set as the default
4. When pressing `Resize Linode` you should see a confirmation dialog appear before resize takes effect
5. During resize we should see resize events as outlined in screenshot and notification (notice) on Linode detail page.
    1. Note: The copy for a `warm` resize has not been updating yet (awaiting API), but you should see the payload with the new `migration_type: warm` as shown in screenshot.
6. A `cold` resize does NOT have any additional confirmation
7. Confirm the confirmation dialog closes on error + the resize panel scrolls to the error for a warm migration; confirm that the resize panel scrolls to the error for a cold migration
8. Confirm that errors do not persist upon closing/reopening the resize panel (for both the same linode and a different linode)
9. Confirm that if a linode is offline, default is cold migration + the panel is disabled and a tooltip appears. Confirm that even if you close/reopen the resize panel, the default stays as the cold migration
10. Confirm the feature flag works as expected
11. Confirm the 'Migration Types' label in the migrations panel is not highlighted when choosing between the migration types
